### ISSUE TITLE
Possible race condition in sf::SoundRecorder

### ISSUE THREAD

**Issue by Foaly** [SFML/SFML#1038]
While I worked on the `sf::SoundRecorder` class @Bromeon noticed there might be a race condition, because `m_isCapturing` is accessed from two different threads. Investigation is needed, to see if a mutex is necessary.

**Comment by LaurentGomila** [comment]
While in a perfect world we would avoid this kind of potential issue, I can't imagine how it could cause any bad effect.

**Comment by Foaly** [comment]
I just opened this issue as a reminder. Looking at the code now I agree. While it's not nice to have a variable accessed from two threads, I don't see any harm that could be done here. For SFML 3 we could use a C++11`std::atomic<bool>` to be 100% sure.

**Comment by Bromeon** [comment]
I don't see an immediate issue now either... But this is the kind of thing where the next refactoring introduces an implicit dependency and subtly blows up everything in very rare circumstances :grimacing:  

There are also some re-occurring patterns like:
```cpp 
if (m_isCapturing) 
{ 
    // Stop the capturing thread 
    m_isCapturing = false; 
    m_thread.wait();
```

**Comment by RajanGrewal** [comment]
Just InterlockedExchange it

**Comment by eXpl0it3r** [comment]
Woops, looks like we forgot to close this one.

Superseded and fixed by #1011
